### PR TITLE
EVM Staking - Validator Tables & Selection updates 

### LINKED ISSUE #1894: [BUG] Limit # of selectable validators in staking UI
## Description
The current Tangle EVM staking UI allows users to select an unlimited number of validators for nomination. This has led to transactions being reverted with a `TooManyTargets` error, indicating that users are attempting to nominate more validators than the protocol allows. We should prevent users from exceeding the maximum allowable number of nominated validators to avoid the error.

### Screenshot
Image

### LINKED ISSUE #1925: [TASK] Show both active and inactive validators in validator list
### Overview

Currently, in our staking interface, when nominators wish to nominate validators, the system displays only a list of active validators to choose from. This results in a scenario where every validator nominated by a user is always in an `active` state. We should display both `active` and `inactive` validators for nominators to nominate.

### Current flow

https://github.com/webb-tools/webb-dapp/assets/53374218/cdad201e-17d5-47bc-b2aa-7f639dc2fcf6

### PR DESCRIPTION
**Pull request by devpavan04**
## Summary of changes

- Nominators can select validators from both the active and waiting pools to nominate.
- Nominators can sort validators by status when choosing validators to nominate.
- An alert message is shown to nominators if they select validators exceeding the maximum nomination quota.
- A table status is added for active and waiting validators. For example, if there are no validators in the waiting pool, the waiting validators table will display a "no validators in waiting pool" status.

### Proposed area of change

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [x] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [ ] `libs/webb-ui-components`

### Reference issue to close (if applicable)

- Closes #1925 
- Closes #1894 

### Screenshots

CleanShot 2024-01-03 at 06 45 41

CleanShot 2024-01-03 at 06 44 30

CleanShot 2024-01-03 at 06 44 17

### CODE REVIEW

**Comment by drewstone** [comment]
Can you provide a video of test/QA?

**Comment by devpavan04** [comment]
@drewstone 


https://github.com/webb-tools/webb-dapp/assets/53374218/3f956747-69be-4e29-a28a-3fc296cfb313

**Comment by devpavan04** [comment]
@drewstone Currently on the testnet, there are zero validators in the waiting pool. This is why we can't see any waiting validators in the above screen recording. I was able to setup a validator locally and test it. The video below demonstrates the working changes. Thanks to @1xstj for helping me run the validator locally ❤️

### First-time nomination flow

https://github.com/webb-tools/webb-dapp/assets/53374218/9e798dd9-4952-4ca9-baff-d1e8f5974c70

### Second time nomination flow

https://github.com/webb-tools/webb-dapp/assets/53374218/9e56558c-c1a8-4dcc-879f-d8750e353aa0

File: apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx [hunk 1]
```diff
@@ -49,6 +51,24 @@ const DelegateTxContainer: FC<DelegateTxContainerProps> = ({
...
+  const { data: waitingValidatorsData } = useSWR(
+    [getWaitingValidators.name],
+    ([, ...args]) => getWaitingValidators(...args)
+  );
+
+  const [maxNominationQuota, setMaxNominationQuota] = useState<number>(0);
+
+  useEffect(() => {
+    getMaxNominationQuota().then((maxNominationQuota) => {
+      setMaxNominationQuota(maxNominationQuota ? maxNominationQuota : 16);
```
Comment (AtelyPham) [inline apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx:63]: The variable `maxNominationQuota` can potentially be `undefined`, `0`, or `NaN`. I recommend implementing a more stringent check for this scenario.

File: apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx [hunk 2]
```diff
@@ -49,6 +51,24 @@ const DelegateTxContainer: FC<DelegateTxContainerProps> = ({
...
+    });
+  }, []);
+
+  const allValidators = useMemo(() => {
+    if (!activeValidatorsData || !waitingValidatorsData) return [];
+
+    return [...activeValidatorsData, ...waitingValidatorsData];
```
Comment (AtelyPham) [inline apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx:70]: Is there any case where we have duplicate validators? Just out of curiosity.

File: apps/tangle-dapp/containers/UpdateNominationsTxContainer/UpdateNominationsTxContainer.tsx [hunk 3]
```diff
@@ -29,6 +34,24 @@ const UpdateNominationsTxContainer: FC<UpdateNominationsTxContainerProps> = ({
...
   );
+  const { data: waitingValidatorsData } = useSWR(
+    [getWaitingValidators.name],
+    ([, ...args]) => getWaitingValidators(...args)
+  );
+
+  const [maxNominationQuota, setMaxNominationQuota] = useState<number>(0);
+
+  useEffect(() => {
+    getMaxNominationQuota().then((maxNominationQuota) => {
```
Comment (AtelyPham) [inline apps/tangle-dapp/containers/UpdateNominationsTxContainer/UpdateNominationsTxContainer.tsx:45]: Same here.

File: apps/tangle-dapp/containers/UpdateNominationsTxContainer/UpdateNominationsTxContainer.tsx [hunk 4]
```diff
@@ -29,6 +34,24 @@ const UpdateNominationsTxContainer: FC<UpdateNominationsTxContainerProps> = ({
...
+      setMaxNominationQuota(maxNominationQuota ? maxNominationQuota : 16);
+    });
+  }, []);
+
+  const allValidators = useMemo(() => {
+    if (!activeValidatorsData || !waitingValidatorsData) return [];
+
+    return [...activeValidatorsData, ...waitingValidatorsData];
+  }, [activeValidatorsData, waitingValidatorsData]);
```
Comment (AtelyPham) [inline apps/tangle-dapp/containers/UpdateNominationsTxContainer/UpdateNominationsTxContainer.tsx:54]: Could you please deduplicate this code? You can achieve this by creating a custom hook to encapsulate the logic. Afterwards, ensure to reuse the hook in both this file and `apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx`.

**Review by AtelyPham — COMMENTED** [review]
Left some comments.

File: apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx (hunk 2 above)
Comment (devpavan04) [reply apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx]: No there won't be.

**Review by drewstone — APPROVED** [review]

**Pull request merged**